src/main/java/com/google/step/youtube/YoutubeInfoScraper.java

```diff
@@ -27,52 +27,68 @@
+      if (response.getItems() == null || response.getItems().isEmpty()) {
+          return Optional.empty();
+      }
+      final List<Channel> channelsInfo = response.getItems();
+      // Since we are only requesting one channel-id, we only get one item back.
+      return Optional.of(channelsInfo.get(0).getContentDetails().getRelatedPlaylists().getUploads());
+  }
+
+  /**
+   * @param uploadId Id of a channel's upload playlist.
+   * @return an optional list of PlaylistItems. Each item contains a video-id,
+   *         description and a date. The optional will be empty if id is invalid
+   *         or no items were found.
+   */
+  public Optional<List<PlaylistItem>> scrapePlaylistItems(final String uploadId) throws IOException {
+      final PlaylistItemListResponse response = youTubeClient.playlistItems().list("snippet").setMaxResults(50L)
+              .setPlaylistId(uploadId).execute();
+      // getItems() return null when no items match the criteria (uploadId).
+      if (response.getItems() == null || response.getItems().isEmpty()) {
+          return Optional.empty();
+      }
+      return Optional.of(response.getItems());
+  }
 
   /**
-  * @param uploadId Id of a channel's upload playlist.
-  * @return an optional list of PlaylistItems. Each item contains a video-id, description and a date. 
-  * The optional will be empty if id is invalid or no items were found.
-  */
-   public Optional<List<PlaylistItem>> scrapePlaylistItems(String uploadId)
-    throws IOException {
-       PlaylistItemListResponse response = youTubeClient.playlistItems().list("snippet")
-            .setMaxResults(50L)
-            .setPlaylistId(uploadId)
-            .execute();
-        // getItems() return null when no items match the criteria (uploadId). 
-        if (response.getItems()==null || response.getItems().isEmpty()) {
-            return Optional.empty();
+   * @param uploadId Id of a channel's upload playlist.
+   * @return an optional list of PromoCode objects. the optional will return empty
+   *         if id is invalid or no items in the playlist are found.
+   */
+  public Optional<List<PromoCode>> scrapePromoCodesFromPlaylist(final String uploadId) throws IOException {
+      final Optional<List<PlaylistItem>> playlistItems = scrapePlaylistItems(uploadId);
+      if (playlistItems.isEmpty()) {
+          return Optional.empty();
+      }
+      final List<PromoCode> promoCodes = new ArrayList<>();
+      for (final PlaylistItem item : playlistItems.get()) {
+          final PlaylistItemSnippet snippet = item.getSnippet();
+          final List<String> itemPromoCodes = DescriptionParser.parse(snippet.getDescription());
+          for (final String promocode : itemPromoCodes) {
+                promoCodes.add(PromoCode.create(promocode, snippet.getResourceId().getVideoId(), 
+                    new Date(snippet.getPublishedAt().getValue())));
+            }
         }
```

Comment:
nit: closing curly braces are not aligned with the for loop declaration columns

Reply:
Fixed.